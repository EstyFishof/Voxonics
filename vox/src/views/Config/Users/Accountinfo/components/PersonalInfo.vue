<template>
  <div class="personal-info-wrap" :class="{'non-pointer-events': roleAgent === 'agent'}">
    <div class="wrap-body">
      <span class="file-loader-span">
        <FileLoader :userUidProp="true" />
      </span>
      <div class="personal-info account-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <label>
              Name*
              <input
                @input="changeUserInfo({ name: userInfo.name }); checkName()"
                placeholder="Enter user name"
                class="input-field"
                maxlength="16"
                v-model.trim="userInfo.name"
                :class="{ error: !userInfo.name.length }"
              />
            </label>
            <div v-if="errors.name" class="alert">{{ errors.name }}</div>
          </div>
          <div class="item-container with-err">
            <label
              >Phone
              <input
                placeholder="Only digits"
                class="input-field"
                type="text"
                v-model.trim="userInfo.phone"
                maxlength="15"
                @input="changeUserInfo({ phone: userInfo.phone || null }); checkPhone()"
              />
            </label>
            <div v-if="errors.phone" class="alert">{{ errors.phone }}</div>
          </div>
          <div class="item-container">
            <label class="options-label" ref="list" v-click-outside="hide"
              >Location:
              <input
                class="input-field num"
                type="text"
                @input="search($event.target.value)"
                v-model="userInfo.location"
                placeholder="London, UK"
                maxlength="16"
              />
              <span
                @click="setParam('showList', 'location', 'location')"
                class="triangle"
                :class="{ 'rotate-arrow': showList === 'location' }"
              >
                <img src="@/assets/img/icons/select-arrow.svg" alt="select" />
              </span>
              <DropdownList v-show="showList === 'location'" :dataList="getGlobalLocations"
                @clickFunc="(item) => {setParam('userInfo.location', item.location); changeUserInfo({ location: userInfo.location || null }) }" 
                :clickFuncAll="null" :classNameList="'list-related-numbers'"  @scrollFunc="(event) => funScroll(event)" :itemText="(item) => item.location" 
                :classNameItemComputed="(item) => userInfoVuex.location === item.location ? 'selected' : ''" />
            </label>
          </div>
          <div class="item-container with-err">
            <label
              >Agent Number (id)*
              <input
                placeholder="000"
                class="input-field"
                type="text"
                v-facade="'###'"
                @input="changeUserInfo({ internalNumber: userInfo.internalNumber }); checkInternalNumber()"
                v-model.trim="userInfo.internalNumber"
                :class="{ error: errors.internalNumber }"
                maxlength="3"
              />
            </label>
            <div v-if="errors.internalNumber" class="alert">
              {{ errors.internalNumber }}
            </div>
          </div>
          <div class="item-container with-err">
            <label
              >Email
              <input
                @input="changeUserInfo({ email: userInfo.email || null }); checkEmail()"
                placeholder="example@voxonics.com"
                class="input-field"
                type="email"
                maxlength="36"
                v-model.trim="userInfo.email"
              />
            </label>
            <div v-if="errors.email" class="alert">{{ errors.email }}</div>
          </div>
          <div class="personal-info__second-part">
            <div class="item-container">
              <label
                >Position
                <input
                  @input="changeUserInfo({ position: userInfo.position || null })"
                  placeholder="Sales Manager"
                  class="input-field"
                  type="text"
                  maxlength="16"
                  v-model.trim="userInfo.position"
                />
              </label>
            </div>
          </div>
          <div class="hr"></div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import FileLoader from "@/components/FileLoader"
import DropdownList from '@/components/DropdownList'
import Validate from "@/lib/validate"
import { mapGetters } from "vuex"
import Api from "@/API/Global.api.js"
import store from "../../../../../store"
import _ from "lodash"
import ClickOutside from "@/lib/clickDirective"

export default {
  name: "PersonalInfo",
  components: {
    FileLoader,
    DropdownList
  },
  data() {
    return {
      unsubscribe: () => {},
      userInfo: {
        name: "",
        phone: "",
        location: "",
        position: "",
        email: "",
        internalNumber: ""
      },
      errors: {
        name: "",
        phone: "",
        email: "",
        internalNumber: "",
        password: "",
        confirm: ""
      },
      count: 10,
      showList: "",
      perPage: 9,
      timer: null
    }
  },
  computed: {
    ...mapGetters(["userInfoVuex", "userInfoComponent", "roleAgent", "getGlobalLocations"])
  },
  methods: {
    funScroll({ target }) {
      if (target.clientHeight + target.scrollTop === target.scrollHeight) {
        if (this.count > this.perPage) {
          this.perPage += 9
          this.fetch("location", this.perPage, this.userInfo.location)
        }
      }
    },
    search(paramSearch) {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch("location", this.perPage, paramSearch), 800)
    },
    checkName() {
      Validate.checkName(this)
    },
    checkPhone() {
      Validate.checkPhone(this)
    },
    checkEmail() {
      Validate.checkEmail(this)
    },
    checkInternalNumber() {
      const num = window.location.href.split("/").pop()
      Validate.checkInternalNumber(this, num)
    },
    checkPassword() {
      Validate.checkPassword(this)
    },
    confirmPassword() {
      Validate.confirmPassword(this)
    },
    fetch(fetchParam, perPage, value) {
      this.$store.dispatch("setSpinner", true)
      switch (fetchParam) {
        case "location":
          return Api.getLocation(perPage, value).then(data => {
            this.count = data.paginator.count
            return data.data
          })
        default:
          break
      }
    },
    setParam(param, value, fetchParam) {
      if (_.get(this, param) === value) {
        _.set(this, param, "")
      } else {
        _.set(this, param, value)
      }
      if (fetchParam) {
        this.fetch(fetchParam, this.perPage, this.userInfo.location)
      }
    },
    changeUserInfo(info) {
      this.showList = ""
      if (this.userInfoComponent.permission.role !== "agent") {
        this.checkInternalNumber()
        this.$store.dispatch("changeUserInfo", info)
      }
    },
    hide() {
      this.showList = ""
    }
  },
  mounted() {
    if (this.roleAgent !== "agent") {
      //Если есть записи в сторе, переносим их в компонент
      for (const key in this.userInfoVuex) {
        this.userInfo[key] = this.userInfoVuex[key]
      }
      this.unsubscribe = store.subscribe(mutation => {
        if (mutation.type === "setUserInfo") {
          for (const key in this.userInfoVuex) {
            this.userInfo[key] = this.userInfoVuex[key]
          }
        }
      })
    } else {
      for (const key in this.userInfoComponent) {
        this.userInfo[key] = this.userInfoComponent[key]
      }
    }
  },
  beforeDestroy() {
    this.unsubscribe()
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss">
.personal-info-wrap {
  background-color: var(--substrate);
  border-radius: 8px 8px 0 0;
  // border: 2px solid var(--cdr-border-row);
  border-bottom: none;

  &.non-pointer-events {
    pointer-events: none;
  }

  .wrap-body {
    display: flex;
    flex-direction: row;
    padding: 0 0 36px 36px;

    .file-loader-span {
      margin: 24px;
    }

    .personal-info {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      justify-content: space-between;
      align-items: center;
      margin-left: 36px;

      .hr {
        width: 100%;
        height: 2px;
        background: var(--cdr-border-row);
        border-radius: 2px;
        flex-grow: 1;
        margin: 32px 24px 0 24px;
      }

      &__first-part,
      &__second-part {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        align-content: flex-start;
        justify-content: flex-start;

        .item-container {
          margin: 16px 24px;
          position: relative;

          label {
            font-size: 18px;
            font-weight: 700;
            text-align: left;
            color: var(--light-blue);
          }

          .error {
            border: 1px solid red;
          }

          .alert {
            position: absolute;
            font-size: 16px;
            font-weight: 300;
            color: #ff0000;
            bottom: -24px;
          }

          .options-label {
            .input-field {
              margin-top: 12px;
              display: block;
              padding: 5px 16px;
              width: 292px;
              height: 35px;
              background: var(--input-bg);
              box-shadow: 0 8px 16px rgba(0, 0, 0, 0.12);
              border-radius: 4px;
              color: var(--font-panel);
              border: none;
            }

            .triangle {
              height: 35px;
              width: calc(100% - 7px);
            }

            .list-related-numbers {
              max-height: 200px;
              overflow-y: auto;
            }
          }
        }
      }
    }
  }

  .with-err {
    position: relative;
    margin-bottom: 30px;
  }
}
@media (max-width: 1408px) {
  .input-field {
    width: 270px;
  }
}

</style>
<style lang="scss">
@media (max-width: 1408px) {
  .personal-info-wrap {
    .wrap-body {
      .account-info {
        .personal-info {
          &__first-part {
            .item-container {
              .options-label {
                .input-field {
                  width: 270px !important;
                }
              }
            }
          }
        }
      }
    }
  }
}
</style>
