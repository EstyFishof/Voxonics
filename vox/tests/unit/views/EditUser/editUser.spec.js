import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import InputFacade from 'vue-input-facade';
import EditUser from '@/views/Config/Users/EditUser/EditUser';
import PersonalInfo from '@/views/Config/Users/EditUser/components/PersonalInfo';
import user from './__mock__/user';
import VueTippy, { TippyComponent } from "vue-tippy";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
localVue.directive('facade', {
    inserted: function (el) {
        const e = el;
    }
});
const router = new VueRouter();

describe('edit user', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters;
    let store;

    beforeEach(() => {
        getters = {
            userInfoVuex: () => user,
            userUid: () => '123',
            changeFields: () => [],
            theme: () => 'dark'
        };
        store = new Vuex.Store({
            getters
        });
    });
    it('should  Vue instance', () => {
        const wrapper = shallowMount(EditUser, { localVue, router, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        const wrapper = shallowMount(EditUser, { localVue, router, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('vuex defined', () => {
        const wrapper = shallowMount(EditUser, { localVue, router, store });
        expect(wrapper.vm.userInfoVuex).toBeDefined();
        expect(wrapper.vm.userUid).toBeDefined();
    });
    it('handlerSubmit defined', () => {
        const wrapper = shallowMount(EditUser, { localVue, router, store });
        expect(wrapper.vm.handlerSubmit).toBeDefined();
    });
    it('handlerSubmit called', () => {
        const wrapper = shallowMount(EditUser, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'handlerSubmit');
        wrapper.vm.handlerSubmit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

});

describe('personal info in editUser', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters, actions;
    let store;
    beforeEach(() => {
        getters = {
            userInfoVuex: () => user,
            theme: () => 'dark'
        };
        actions = {
            setUserInfo: jest.fn(),
            changeUserInfo: jest.fn()
        };
        store = new Vuex.Store({
            actions,
            getters
        });
    });
    it('should Vue instance', () => {
        const wrapper = shallowMount(PersonalInfo, {
            localVue, router, store, directives: {
                facade: el => {
                }
            }
        });
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('userInfoVuex defined', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        expect(wrapper.vm.userInfoVuex).toBeDefined();
    });
    it('checkName ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkName');
        wrapper.vm.checkName(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('checkPhone', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkPhone');
        wrapper.vm.checkPhone(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('checkEmail', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkEmail');
        wrapper.vm.checkEmail(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('checkInternalNumber ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkInternalNumber');
        wrapper.vm.checkInternalNumber(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('checkPassword ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkPassword');
        wrapper.vm.checkPassword(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('confirmPassword ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'confirmPassword');
        wrapper.vm.confirmPassword(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('switchVisibility ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'switchVisibility');
        wrapper.vm.switchVisibility();
        expect(spy).toHaveBeenCalled();
    });
    it('switchVisibilityConfirm ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'switchVisibilityConfirm');
        wrapper.vm.switchVisibilityConfirm();
        expect(spy).toHaveBeenCalled();
    });
    it('setUserInfo ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'setUserInfo');
        wrapper.vm.setUserInfo('info');
        expect(spy).toHaveBeenCalled();
    });
});
