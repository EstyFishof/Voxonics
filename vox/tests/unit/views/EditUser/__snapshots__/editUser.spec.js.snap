// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit user Check snapshot 1`] = `
<div
  class="wrap-main"
>
  <deleteusers-stub />
   
  <div
    class="header"
  >
    <div
      class="tab"
    >
      <svg
        fill="none"
        height="29"
        viewBox="20 15 30 30"
        width="33"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#17212B"
          height="29"
          rx="8"
          width="33"
          x="17"
          y="16"
        />
         
        <mask
          fill="white"
          id="path-2-inside-1"
        >
          <path
            clip-rule="evenodd"
            d="M45.9469 39.0923C45.9469 39.6852 45.4293 40.1663 44.789 40.1663C44.1486 40.1663 43.6311 39.6852 43.6311 39.0923C43.6311 36.1311 41.0339 33.7219 37.8416 33.7219C34.6493 33.7219 32.0521 36.1311 32.0521 39.0923C32.0521 39.6852 31.5345 40.1663 30.8942 40.1663C30.2539 40.1663 29.7363 39.6852 29.7363 39.0923C29.7363 34.9464 33.3733 31.5738 37.8416 31.5738C42.3099 31.5738 45.9469 34.9464 45.9469 39.0923ZM37.8416 23.25C39.1187 23.25 40.1574 24.3339 40.1574 25.6667C40.1574 26.9995 39.1187 28.0833 37.8416 28.0833C36.5644 28.0833 35.5258 26.9995 35.5258 25.6667C35.5258 24.3339 36.5644 23.25 37.8416 23.25ZM37.8425 30.5007C40.3968 30.5007 42.4741 28.3329 42.4741 25.6673C42.4741 23.0017 40.3968 20.834 37.8425 20.834C35.2882 20.834 33.2109 23.0017 33.2109 25.6673C33.2109 28.3329 35.2882 30.5007 37.8425 30.5007Z"
            fill-rule="evenodd"
          />
        </mask>
         
        <path
          clip-rule="evenodd"
          d="M45.9469 39.0923C45.9469 39.6852 45.4293 40.1663 44.789 40.1663C44.1486 40.1663 43.6311 39.6852 43.6311 39.0923C43.6311 36.1311 41.0339 33.7219 37.8416 33.7219C34.6493 33.7219 32.0521 36.1311 32.0521 39.0923C32.0521 39.6852 31.5345 40.1663 30.8942 40.1663C30.2539 40.1663 29.7363 39.6852 29.7363 39.0923C29.7363 34.9464 33.3733 31.5738 37.8416 31.5738C42.3099 31.5738 45.9469 34.9464 45.9469 39.0923ZM37.8416 23.25C39.1187 23.25 40.1574 24.3339 40.1574 25.6667C40.1574 26.9995 39.1187 28.0833 37.8416 28.0833C36.5644 28.0833 35.5258 26.9995 35.5258 25.6667C35.5258 24.3339 36.5644 23.25 37.8416 23.25ZM37.8425 30.5007C40.3968 30.5007 42.4741 28.3329 42.4741 25.6673C42.4741 23.0017 40.3968 20.834 37.8425 20.834C35.2882 20.834 33.2109 23.0017 33.2109 25.6673C33.2109 28.3329 35.2882 30.5007 37.8425 30.5007Z"
          fill="#276899"
          fill-rule="evenodd"
        />
         
        <path
          d="M43.9469 39.0923C43.9469 38.4429 44.4676 38.1663 44.789 38.1663V42.1663C46.391 42.1663 47.9469 40.9274 47.9469 39.0923H43.9469ZM44.789 38.1663C45.1104 38.1663 45.6311 38.4429 45.6311 39.0923H41.6311C41.6311 40.9274 43.1869 42.1663 44.789 42.1663V38.1663ZM45.6311 39.0923C45.6311 34.8876 41.9943 31.7219 37.8416 31.7219V35.7219C40.0735 35.7219 41.6311 37.3745 41.6311 39.0923H45.6311ZM37.8416 31.7219C33.6889 31.7219 30.0521 34.8876 30.0521 39.0923H34.0521C34.0521 37.3745 35.6096 35.7219 37.8416 35.7219V31.7219ZM30.0521 39.0923C30.0521 38.4429 30.5728 38.1663 30.8942 38.1663V42.1663C32.4963 42.1663 34.0521 40.9274 34.0521 39.0923H30.0521ZM30.8942 38.1663C31.2156 38.1663 31.7363 38.4429 31.7363 39.0923H27.7363C27.7363 40.9274 29.2922 42.1663 30.8942 42.1663V38.1663ZM31.7363 39.0923C31.7363 36.19 34.3335 33.5738 37.8416 33.5738V29.5738C32.4131 29.5738 27.7363 33.7027 27.7363 39.0923H31.7363ZM37.8416 33.5738C41.3497 33.5738 43.9469 36.19 43.9469 39.0923H47.9469C47.9469 33.7027 43.2701 29.5738 37.8416 29.5738V33.5738ZM37.8416 25.25C37.8982 25.25 37.9661 25.271 38.0344 25.3423C38.104 25.415 38.1574 25.5261 38.1574 25.6667H42.1574C42.1574 23.3105 40.3028 21.25 37.8416 21.25V25.25ZM38.1574 25.6667C38.1574 25.8072 38.104 25.9184 38.0344 25.991C37.9661 26.0623 37.8982 26.0833 37.8416 26.0833V30.0833C40.3028 30.0833 42.1574 28.0228 42.1574 25.6667H38.1574ZM37.8416 26.0833C37.785 26.0833 37.7171 26.0623 37.6488 25.991C37.5792 25.9184 37.5258 25.8072 37.5258 25.6667H33.5258C33.5258 28.0228 35.3804 30.0833 37.8416 30.0833V26.0833ZM37.5258 25.6667C37.5258 25.5261 37.5792 25.415 37.6488 25.3423C37.7171 25.271 37.785 25.25 37.8416 25.25V21.25C35.3804 21.25 33.5258 23.3105 33.5258 25.6667H37.5258ZM37.8425 32.5007C41.5809 32.5007 44.4741 29.3562 44.4741 25.6673H40.4741C40.4741 27.3096 39.2127 28.5007 37.8425 28.5007V32.5007ZM44.4741 25.6673C44.4741 21.9784 41.5809 18.834 37.8425 18.834V22.834C39.2127 22.834 40.4741 24.0251 40.4741 25.6673H44.4741ZM37.8425 18.834C34.1041 18.834 31.2109 21.9784 31.2109 25.6673H35.2109C35.2109 24.0251 36.4723 22.834 37.8425 22.834V18.834ZM31.2109 25.6673C31.2109 29.3562 34.1041 32.5007 37.8425 32.5007V28.5007C36.4723 28.5007 35.2109 27.3096 35.2109 25.6673H31.2109Z"
          fill="#276899"
          mask="url(#path-2-inside-1)"
        />
         
        <path
          clip-rule="evenodd"
          d="M29.1942 30.2363H25.9376V26.8379C25.9376 26.5247 25.6944 26.2715 25.3948 26.2715C25.0952 26.2715 24.8521 26.5247 24.8521 26.8379V30.2363H21.5955C21.2959 30.2363 21.0527 30.4895 21.0527 30.8027C21.0527 31.116 21.2959 31.3691 21.5955 31.3691H24.8521V34.7676C24.8521 35.0808 25.0952 35.334 25.3948 35.334C25.6944 35.334 25.9376 35.0808 25.9376 34.7676V31.3691H29.1942C29.4938 31.3691 29.7369 31.116 29.7369 30.8027C29.7369 30.4895 29.4938 30.2363 29.1942 30.2363Z"
          fill="#276899"
          fill-rule="evenodd"
        />
         
        <defs>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="61"
            id="filter0_d"
            width="65"
            x="0"
            y="0"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
             
            <fecolormatrix
              in="SourceAlpha"
              type="matrix"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            />
             
            <feoffset
              dx="-1"
            />
             
            <fegaussianblur
              stdDeviation="8"
            />
             
            <fecolormatrix
              type="matrix"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.15 0"
            />
             
            <feblend
              in2="BackgroundImageFix"
              mode="normal"
              result="effect1_dropShadow"
            />
             
            <feblend
              in="SourceGraphic"
              in2="effect1_dropShadow"
              mode="normal"
              result="shape"
            />
          </filter>
           
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="27.3323"
            id="filter1_d"
            width="24.2105"
            x="25.7363"
            y="20.834"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
             
            <fecolormatrix
              in="SourceAlpha"
              type="matrix"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            />
             
            <feoffset
              dy="4"
            />
             
            <fegaussianblur
              stdDeviation="2"
            />
             
            <fecolormatrix
              type="matrix"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
            />
             
            <feblend
              in2="BackgroundImageFix"
              mode="normal"
              result="effect1_dropShadow"
            />
             
            <feblend
              in="SourceGraphic"
              in2="effect1_dropShadow"
              mode="normal"
              result="shape"
            />
          </filter>
           
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="17.0625"
            id="filter2_d"
            width="16.6842"
            x="17.0527"
            y="26.2715"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
             
            <fecolormatrix
              in="SourceAlpha"
              type="matrix"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            />
             
            <feoffset
              dy="4"
            />
             
            <fegaussianblur
              stdDeviation="2"
            />
             
            <fecolormatrix
              type="matrix"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
            />
             
            <feblend
              in2="BackgroundImageFix"
              mode="normal"
              result="effect1_dropShadow"
            />
             
            <feblend
              in="SourceGraphic"
              in2="effect1_dropShadow"
              mode="normal"
              result="shape"
            />
          </filter>
        </defs>
      </svg>
       
      <span>
        Settings overview
      </span>
    </div>
  </div>
   
  <img
    alt="trash"
    class="trash"
    content="Delete user"
    src="../../../../assets/img/icons/trash.svg"
    width="30"
  />
   
  <div
    class="wrap-form"
  >
    <form>
      <personalinfo-stub />
       
      <agentsetup-stub />
       
      <qsettings-stub />
       
      <managersettings-stub />
       
      <div
        class="btn-container"
      >
        <button
          class="btn-submit"
          disabled="disabled"
        >
          Apply Changes
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`personal info in editUser Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      PERSONAL INFO
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="wrap-body"
  >
    <fileloader-stub />
     
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <label>
            
            Name*
            
            <input
              class="input-field error"
              maxlength="32"
              placeholder="Enter user name"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container with-err"
        >
          <label>
            Phone
            
            <input
              class="input-field"
              maxlength="15"
              placeholder="Only digits"
              type="text"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container"
        >
          <label>
            Location
            
            <input
              class="input-field"
              placeholder="London, UK"
              type="text"
            />
          </label>
        </div>
         
        <div
          class="item-container with-err"
        >
          <label>
            Agent Number (id)*
            
            <input
              class="input-field"
              placeholder="000"
              type="text"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container with-err"
        >
          <label>
            Email
            
            <input
              class="input-field"
              maxlength="36"
              placeholder="example@voxonics.com"
              type="email"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container"
        >
          <label>
            Position
            
            <input
              class="input-field"
              maxlength="16"
              placeholder="Sales Manager"
              type="text"
            />
          </label>
        </div>
         
        <div
          class="item-container password-field"
        >
          <label>
            Password*
            
            <div
              class="icon-eye icon-eye-close"
            />
             
            <input
              class="input-field"
              placeholder="Enter password"
              type="password"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container password-field"
        >
          <label>
            Confirm*
            
            <div>
              <span
                class="icon-eye-close"
              />
            </div>
             
            <input
              class="input-field"
              placeholder="Confirm password"
              type="password"
            />
             
            <!---->
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
